### PR TITLE
Add owasp dependency check.

### PR DESCRIPTION
**Pull request by suryatej16**
###### Problem:
Currently, the dropwizard project does not run a check for security vulnerabilities during its build process. This results in dependencies being included that contain security risks (e.g. Jackson versions below 2.9.5)

###### Solution:
I have added the dependency-check plugin to the maven build process (https://jeremylong.github.io/DependencyCheck/) so CVEs can be identified.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -53,6 +53,8 @@
 
     <properties>
         <dropwizard.url>http://www.dropwizard.io/${project.version}</dropwizard.url>
+        <dropwizard.check.fail-owasp>true</dropwizard.check.fail-owasp>
+        <dropwizard.check.skip-owasp>false</dropwizard.check.skip-owasp>
```
Comment (joschi) [inline pom.xml]: No need to introduce another property here. The OWASP dependency check plugin provides the `dependency-check.skip` property out of the box with the same default.

File: pom.xml [hunk 2]
```diff
@@ -53,6 +53,8 @@
 
     <properties>
         <dropwizard.url>http://www.dropwizard.io/${project.version}</dropwizard.url>
+        <dropwizard.check.fail-owasp>true</dropwizard.check.fail-owasp>
```
Comment (joschi) [inline pom.xml]: No need to introduce another property here. The OWASP dependency check plugin provides the `failOnError` and `failBuildOnAnyVulnerability` properties out of the box.

File: pom.xml [hunk 3]
```diff
@@ -665,6 +667,27 @@
...
+
+            <plugin>
+                <groupId>org.owasp</groupId>
+                <artifactId>dependency-check-maven</artifactId>
+                <version>3.1.2</version>
+                <configuration>
+                    <failBuildOnAnyVulnerability>${dropwizard.check.fail-owasp}</failBuildOnAnyVulnerability>
+                    <skip>${dropwizard.check.skip-owasp}</skip>
+                    <showSummary>true</showSummary>
+                    <format>ALL</format>
```
Comment (joschi) [inline pom.xml]: Do we really need all report formats here? I think HTML (default) should be sufficient.

https://jeremylong.github.io/DependencyCheck/dependency-check-maven/configuration.html

**Comment by nickbabcock** [comment]
I find it bizarre that the jdk10 build doesn't report the same false positive, gets a bit farther in the build, and then reports a different false positive: [CVE-2015-8862](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2015-8862) 😅

**Comment by suryatej16** [comment]
@joschi Added a suppression file to ignore false positives. Just a general question, how did you distinguish those as false positives?

**Comment by joschi** [comment]
> how did you distinguish those as false positives?

@suryatej16 From the fact that Dropwizard is not using the PHP CMS Drupal. 😉

File: suppressed-cves.xml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<suppressions xmlns="https://jeremylong.github.io/DependencyCheck/dependency-suppression.1.1.xsd">
+    <suppress>
+        <notes><![CDATA[file name: dropwizard-views-1.4.0-SNAPSHOT.jar]]></notes>
+        <gav regex="true">^io\.dropwizard:dropwizard-views(-.*)?:.*$</gav>
+        <cve>CVE-2015-3378</cve>
+    </suppress>
+    <suppress>
+        <notes><![CDATA[file name: dropwizard-views-1.4.0-SNAPSHOT.jar]]></notes>
```
Comment (joschi) [inline suppressed-cves.xml]: That's a bit specific for the current (SNAPSHOT) version. Maybe add a comment that it is a false positive instead.

File: suppressed-cves.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<suppressions xmlns="https://jeremylong.github.io/DependencyCheck/dependency-suppression.1.1.xsd">
+    <suppress>
+        <notes><![CDATA[file name: dropwizard-views-1.4.0-SNAPSHOT.jar]]></notes>
```
Comment (joschi) [inline suppressed-cves.xml]: That's a bit specific for the current (SNAPSHOT) version. Maybe add a comment that it is a false positive instead.

File: suppressed-cves.xml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        <gav regex="true">^io\.dropwizard:dropwizard-views(-.*)?:.*$</gav>
```
Comment (joschi) [inline suppressed-cves.xml]: I think it's safe to ignore CVE-2015-3378 completely and not just for this artifact.

File: suppressed-cves.xml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        <cve>CVE-2015-3378</cve>
+    </suppress>
+    <suppress>
+        <notes><![CDATA[file name: dropwizard-views-1.4.0-SNAPSHOT.jar]]></notes>
+        <gav regex="true">^io\.dropwizard:dropwizard-views(-.*)?:.*$</gav>
```
Comment (joschi) [inline suppressed-cves.xml]: I think it's safe to ignore CVE-2015-3379 completely and not just for this artifact.

File: suppressed-cves.xml [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+        <cve>CVE-2015-3379</cve>
+    </suppress>
+    <suppress>
+        <notes><![CDATA[file name: compiler-0.9.5.jar]]></notes>
+        <gav regex="true">^com\.github\.spullara\.mustache\.java:compiler:.*$</gav>
```
Comment (joschi) [inline suppressed-cves.xml]: I think it's safe to ignore CVE-2015-8862 completely and not just for this artifact.

File: suppressed-cves.xml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+        <gav regex="true">^io\.dropwizard:dropwizard-views(-.*)?:.*$</gav>
+        <cve>CVE-2015-3378</cve>
+    </suppress>
+    <suppress>
+        <notes><![CDATA[file name: dropwizard-views-1.4.0-SNAPSHOT.jar]]></notes>
+        <gav regex="true">^io\.dropwizard:dropwizard-views(-.*)?:.*$</gav>
+        <cve>CVE-2015-3379</cve>
+    </suppress>
+    <suppress>
+        <notes><![CDATA[file name: compiler-0.9.5.jar]]></notes>
```
Comment (joschi) [inline suppressed-cves.xml]: That's a bit specific for the version of the Mustache template engine currently used by Dropwizard. Maybe add a comment that it is a false positive instead

**Comment by suryatej16** [comment]
Updated with review comments.

File: suppressed-cves.xml [hunk 10]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<suppressions xmlns="https://jeremylong.github.io/DependencyCheck/dependency-suppression.1.1.xsd">
+    <suppress>
+        <notes><![CDATA[false positive, dropwizard does not use Drupal]]></notes>
+        <cve>CVE-2015-3378</cve>
+    </suppress>
+    <suppress>
+        <notes><![CDATA[false positive, dropwiard does not use Drupal]]></notes>
```
Comment (joschi) [inline suppressed-cves.xml]: Typo: dropwiard -> Dropwizard

File: suppressed-cves.xml [hunk 11]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<suppressions xmlns="https://jeremylong.github.io/DependencyCheck/dependency-suppression.1.1.xsd">
+    <suppress>
+        <notes><![CDATA[false positive, dropwizard does not use Drupal]]></notes>
```
Comment (joschi) [inline suppressed-cves.xml]: Typo: dropwiard -> Dropwizard

**Pull request merged**